### DISCUSSION TITLE
EDEGX TAF testcase failure in not consistent

### DISCUSSION THREAD

**Discussion by divyashree123456** [edgexfoundry/community#151] [Q&A] [answered]
Hi @cloudxxx8 , @chr1shung 
We are testing TAF test-cases on riscv machine with our own images for edgex services and for taf ,while testing  failure of test case are varying.[Jakata version]
for example : we are testing core-data ,for which passed test-cases are some times 54, 51,49,52 outof 56 testcases.
why there is  inconsistency  ? Need to change any configuration.

please do  help me to resolve this with your inputs 
Thanks in adavance
![Screenshot from 2023-07-20 13-57-55](https://github.com/edgexfoundry/community/assets/120623642/ba5710b8-4a20-47ca-83e7-afef106581e5)
![Screenshot from 2023-07-20 14-06-20](https://github.com/edgexfoundry/community/assets/120623642/0161d83c-b79c-42df-9ac3-727dc5aadfcf)
![Screenshot from 2023-07-20 14-16-47](https://github.com/edgexfoundry/community/assets/120623642/d6567610-7476-429c-8bf5-fa2f8f7e7bd8)

**Comment by cloudxxx8** [comment]
it requres further investigation in which test cases sometimes fail.
you can use and robot framework debug tool to check it out, and observe the difference when they are pass and fail.

**Comment by divyashree123456** [comment]
Hi @cloudxxx8  
how this debugger help for my inconsistent failure of test cases ? please help me understand better for this debugger.
To use this debugger ,need to modify robot scripts?

**Reply by divyashree123456** [reply]
Hi @cloudxxx8 ,One more clarification for running a Test cases of CORE-DATA is it necessary to run device-virtual service?
Because earlier I was facing the inconsistency while testing the test cases ,sometimes there were failing or passing.but when I tried running test-cases for core-data ,by stoping device-virtual container ,I was getting constitent report of test cases execution.

**Reply by cloudxxx8** [reply, marked as answer]
no, the test should not include device-virtual, because it will send events automatically and break the event count in this test.
please just follow the debployment script for the test.  if it deploys device-virtual, the pre-defined devices should be removed first.

**Reply by divyashree123456** [reply]
Hi @cloudxxx8 ,Thanks for suggestions ,It had helped to to understand and resolve  some issues.

**Reply by divyashree123456** [reply]
Hi @cloudxxx8 ,while trying with deploy scripts also ,some time some of the  test cases are failing and passing .its report is not consistent .
Why is this so?please help me to understand better.(In X86 machine)

**Reply by cloudxxx8** [reply]
The reason could various, such as hardware speed, network enviromnet, or test script bugs.
It should be investigated case by case.
We can just make sure the test report is consistent in the community Jenkins.
